tests/Kernel/CorporateBlocks/CorporateBlocksTestBase.php

```diff
@@ -5,19 +5,161 @@
+    ]);
+    $corporate_site_info->save();
+
+    $config = $this->configFactory->getEditable('system.site');
+    $config->set('name', 'OpenEuropa');
+    $config->save();
+  }
+
+  /**
+   * Render a corporate footer block with test data.
+   *
+   * @param string $type
+   *   The type of block, ec or eu.
+   * @param array $test_data
+   *   The test data for config and assertion.
+   *
+   * @return string
+   *   The rendered HTML.
+   */
+  protected function renderCorporateBlocksFooter(string $type, array &$test_data): string {
+    // Override corporate block footer config with test data.
+    $config_name = "oe_corporate_blocks.{$type}_data.footer";
+    $fixture_name = "{$type}_footer.yml";
+    $block_id = "oe_corporate_blocks_{$type}_footer";
+
+    /* @var $config_obj \Drupal\Core\Config\Config */
+    $config_obj = $this->configFactory->getEditable($config_name);
+    $test_data = $this->getFixtureContent($fixture_name);
+    $config_obj->setData($test_data);
+    $config_obj->save();
+
+    // Setup and render footer block.
+    $config = [
+      'id' => $block_id,
+      'label' => 'OpenEuropa footer block',
+      'provider' => 'oe_corporate_blocks',
+      'label_display' => '0',
+    ];
+    $build = $this->buildBlock($block_id, $config);
+
+    return $this->renderRoot($build);
+  }
+
+  /**
+   * We assigned each section two links, assert that content and order.
+   *
+   * @param \Symfony\Component\DomCrawler\Crawler $section
+   *   The footer section.
+   * @param string $branding
+   *   Ecl branding, core/standardised.
+   * @param array $expected
+   *   The expected data.
+   */
+  protected function assertLinkList(Crawler $section, string $branding, array $expected): void {
+    $actual = $section->filter('ul li:nth-child(1) > a');
+    $this->assertEquals($expected['0']['href'], $actual->attr('href'));
+    $this->assertEquals($expected['0']['label'], $actual->text());
+    $this->assertEquals("ecl-link ecl-link--standalone ecl-footer-{$branding}__link", $actual->attr('class'));
+
+    $actual = $section->filter('ul li:nth-child(2) > a');
+    $this->assertEquals($expected['1']['href'], $actual->attr('href'));
+    $this->assertEquals($expected['1']['label'], $actual->text());
+    $this->assertEquals("ecl-link ecl-link--standalone ecl-footer-{$branding}__link", $actual->attr('class'));
+  }
+
+  /**
+   * Assert footer block is present and has correct number of sections.
+   *
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   Crawler containing the DOM nodes.
+   * @param string $branding
+   *   Ecl branding, core/standardised.
+   * @param int $expected_section_count
+   *   The number of expected sections.
+   */
+  protected function assertFooterPresence(Crawler $crawler, string $branding, int $expected_section_count): void {
+    $actual = $crawler->filter("footer.ecl-footer-{$branding}");
+    $this->assertCount(1, $actual);
```

Comment:
When we are creating assertion methods, let's always add a message that explains why something failed. If we keep the above we will get "0 expected to be 1". Something like:

```php
     $this->assertCount(1, $actual, "Footer with {$branding} branding not found.");
```

Reply:
I switched to assertSession()->elementExists in the functional test which already has these messages implemented.